lib/src/arg_parser.dart

```diff
@@ -29,15 +30,26 @@ class ArgParser {
   /// arguments.
   final bool allowTrailingOptions;
 
+  /// Whether or not this parser treats unrecognized options as non-option
+  /// arguments.
+  bool get allowsAnything => false;
+
   /// Creates a new ArgParser.
   ///
   /// If [allowTrailingOptions] is set, the parser will continue parsing even
   /// after it finds an argument that is neither an option nor a command.
   /// This allows options to be specified after regular arguments. Defaults to
   /// `false`.
   factory ArgParser({bool allowTrailingOptions: false}) =>
-      new ArgParser._(<String, Option>{}, <String, ArgParser>{},
-          allowTrailingOptions: allowTrailingOptions);
+      new ArgParser._({}, {}, allowTrailingOptions: allowTrailingOptions);
+
+  /// Creates a new ArgParser that treats *all input* as non-option arguments.
+  ///
+  /// This is intended to allow arguments to be passed through to child
+  /// processes without needing to be redefined in the parent.
+  ///
+  /// Options may not be defined for this parser.
+  factory ArgParser.allowAnything() = AllowAnythingParser;
```

Comment:
I'm not sure how that would work with the command runner—the user is likely to still want a custom `Command` with a custom `run()` method, which means they need *some* implementation for `Command.argParser`.

More broadly, I think it's useful to give users the ability to slot in a parser that allows anything to any API that expects an `ArgParser`. `CommandRunner` is the most obvious example of such an API right now, but we should design for the possibility that more appear later.